### PR TITLE
fix daemon utils for py3

### PR DESCRIPTION
**Pull request by kwaegema**

### CODE REVIEW

File: lib/vsc/utils/daemon.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def daemonize(self):
         sys.stderr.flush()
         si = open(self.stdin, 'r')
         so = open(self.stdout, 'a+')
-        se = open(self.stderr, 'a+', 0)
+        se = open(self.stderr, 'ba+', 0)
```
Comment (stdweird) [inline lib/vsc/utils/daemon.py:68]: why is it not required for stdout?

Comment (boegel) [reply lib/vsc/utils/daemon.py:68]: Because `stdout` is not being opened unbuffered (that's what the `0` is about).

Comment (kwaegema) [reply lib/vsc/utils/daemon.py:68]: unbuffered (the 0) is only possible with binary, so its either b with 0 or not b and not 0

Comment (stdweird) [reply lib/vsc/utils/daemon.py:68]: again, why is it not the same?

Comment (boegel) [reply lib/vsc/utils/daemon.py:68]: It's not our code, and non-buffered is usually only done for `stderr`?

Comment (stdweird) [reply lib/vsc/utils/daemon.py:68]: yeah, whatever the reason, it seems to be some standard see https://pagure.io/python-daemon/blob/master/f/daemon/runner.py#_132

**Comment by stdweird** [comment]
@kwaegema py2 tests still fail

**Comment by kwaegema** [comment]
@stdweird Hmm, weird, I'm not sure where this is coming from..
```FAIL: test_timeout (test.run.TestRun)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/var/lib/jenkins/jobs/hpcugent_github.com/jobs/vsc-base/branches/PR-311/workspace/test/run.py", line 295, in test_timeout

    do_test(False)

  File "/var/lib/jenkins/jobs/hpcugent_github.com/jobs/vsc-base/branches/PR-311/workspace/test/run.py", line 273, in do_test

    self.assertFalse(check_pid(pids[depth]), "main depth=%s pid (pids %s) is killed by timeout" % (depth, pids,))

AssertionError: True is not false : main depth=2 pid (pids [22371, 22369, 22368]) is killed by timeout
```
It doesn't seem related to my change and I can't reproduce it on my laptop,. Also python tests / python_unittests (2.7) (pull_request)  is ok

**Pull request merged**